Swiftfin/Views/ItemView/ItemView.swift

```diff
@@ -100,6 +127,68 @@ struct ItemView: View {
             if viewModel.backgroundStates.contains(.refresh) {
                 ProgressView()
             }
+            if enableMenu {
+                itemActionMenu
+            }
+        }
+        .confirmationDialog(
+            L10n.deleteItemConfirmationMessage,
+            isPresented: $showConfirmationDialog,
+            titleVisibility: .visible
+        ) {
+            Button(L10n.confirm, role: .destructive) {
+                deleteViewModel.send(.delete)
+            }
+            Button(L10n.cancel, role: .cancel) {}
+        }
+        .onReceive(deleteViewModel.events) { event in
+            switch event {
+            case let .error(eventError):
+                error = eventError
+                isPresentingEventAlert = true
+            case .deleted:
+                if deleteViewModel.item == nil {
```

Comment:
There shouldn't be a situation where we succeed on deleting the item and it wasn't set to `nil`. If it fails, we should have gotten an `error` event. So, we don't need this kind of check - if we get the `deleted` event then it succeeded.